extensions/panache/panache-rest-common/runtime/src/main/java/io/quarkus/panache/rest/runtime/PanacheCrudResource.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.panache.rest.runtime;
```

Comment:
Hmm now that you asked it makes me wonder if this won't confused the users too. Maybe I should get rid of this supertype and duplicate the methods in each interface? Otherwise somebody might try to use this interface directly.

Reply:
Initially I put it in a `runtime` package to indicate that it shouldn't be used directly.

Reply:
Well, why should these methods not be used? Probably this should just be made public.

Reply:
Not the methods, but this interface shouldn't be used directly. The Processor scans for `PanacheEntityCrudResource` and `PanacheRepositoryCrudResource` which extend this interface.

Reply:
And then Mongo would have similar alternatives

Reply:
The type hierarchy is public in Java, so there's no point trying to hide this interface. Own it ;) Unless you want to hide it to avoid errors where the users would implement it directly by mistake, but even then it's still public, so they'll see it just the same and the risk is not reduced.

If that's such a danger, then yes, duplicate the methods in the two interfaces and remove this one, but I really doubt it's so dangerous.

Reply:
OK I agree.  Moving it to the common package.